Create proposal

alexisljn opened the pull request.

alexisljn committed:
Implement The Create Proposal Feature

Setting Up the markdown wysiwyg editor

alexisljn committed:
Get HTML Input From Markdown Editor

After a click of the user

alexisljn committed:
Merge branch 'create-proposal' of https://github.com/alexisljn/collab-media into create-proposal

alexisljn committed:
Create Proposal Form

Form is implemented

alexisljn committed:
Merge branch 'master' into create-proposal

alexisljn committed:
Send Front Content To Back Logic

Using jQuery

alexisljn committed:
Create Body Of New Models

proposal and content proposal

alexisljn committed:
Merge branch 'master' into create-proposal

EmileCalixte committed:
Update docker-compose

Updated docker-compose.yml to use a custom php-nginx image, in order
to increase the max upload size

alexisljn committed:
Temporary Commit To Test Another Features

Coded by Emile.

EmileCalixte committed:
Revert To Base Docker Image And Add Nginx Config Volume Sharing

Reverted to previous image docker, added custom proxy settings for nginx
according to https://github.com/Note-able/noteable/commit/9adbb2a134e9e295e0890633d80b4ae97982ef7f

alexisljn committed:
Push To Work On New Docker Compose

Nginx to Apache

EmileCalixte committed:
Change docker image for Apache webserver

Changed docker webserver image from nginx to apache

EmileCalixte committed:
Merge remote-tracking branch 'origin/create-proposal' into update-docker-compose

EmileCalixte committed:
Increase PHP max file upload size limit

Increased PHP max file upload size limit to 500MB

alexisljn committed:
Finalize Form Logic

Refactorization is needed.

alexisljn committed:
Refactoring Code

Into multiple short private functions

alexisljn committed:
Add PHPDoc To Newest Functions And Create Custom Exception

Exception for the file uploaded by the user on the server.

alexisljn committed:
Remove Of Useless Layout

Tui.editor logic moved into main layout.
Handling in view of error generated by uploaded file.
Cleaning of Controller

EmileCalixte commented on components/Util.php:
```diff
@@ -8,6 +8,8 @@
 
 class Util
 {
+    public const ALLOWED_EXTENSIONS = ['png', 'jpg', 'jpeg', 'gif', 'mp3', 'mp4', 'mov', 'pdf', 'zip' ];
```
Could we name this constant "UPLOADED_FILE_ALLOWED_EXTENSIONS" or something like this ?

alexisljn replied:
Deal !

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
 
+    /**
+     * Display a form to create a Proposal
+     * Save the proposal in DDB when form is submitted
```
DB*

alexisljn replied:
What a shame !

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
+            $transaction = Yii::$app->db->beginTransaction();
+
+            try {
+                $post = Yii::$app->request->post();
```
$post may be ambiguous when we will create the action(s) to post a proposal on a social network. I think you should rename it to $postParams, or more simply just use $_POST which is equivalent.

alexisljn replied:
You're the man ! You always have a decade ahead. I change that into $postRequest

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
+    }
+
+    /**
+     * Save proposal in DDB
```
DB*

alexisljn replied:
What a shame !

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
+        $proposal = new Proposal();
+        $proposal->title = $proposalTitle;
+        $proposal->submitter_id = mainController::getCurrentUser()->id;
+        $proposal->status = 'pending';
```
I think we have to create a Proposal model which extends the base databaseModels/Proposal, create constants for each proposal status, and use the constants instead of using directly the name of the status

alexisljn replied:
Yes, you're right. In case of future change it will be easier to edit the const one time instead of several times in code.

alexisljn committed:
Change For PR Approval

Change post variable into postRequest
DDB into DB
Creation of Proposal Model to declare status const

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
+    }
+
+    /**
+     * Save ProposalContentHistory in DDB
```
DB*

alexisljn replied:
What a shame !

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
+    }
+
+    /**
+     * Save the file in DDB.
```
DB*

alexisljn replied:
What a shame !

alexisljn committed:
Remove Useless Namespace

In models/Proposal

EmileCalixte commented on views/proposal/create-proposal.php:
```diff
@@ -0,0 +1,39 @@
+/** @var string $error  */
+use yii\helpers\Html; ?>
+
+<?= 'CREATE PROPOSAL'; ?>
```
🤔

alexisljn replied:
My bad, I change that into a beautiful h1 tag :D

alexisljn committed:
Change PHP Echo Into H1 Tag

for create proposal heading.

alexisljn committed:
Change Const Name

For the allowed extensions.

EmileCalixte reviewed (approved):
Great !
